### DISCUSSION TITLE
List of incompatible Lenses

### DISCUSSION THREAD

**Discussion by ptrumpis** [ptrumpis/snap-camera-server#8] [General]
## List of Lenses that are known to be incompatible with the latest Snap Camera application.

Some lenses (infinite loading bug) can be fixed with an unofficial update and additional assets.
See here: [Unofficial Update](https://github.com/ptrumpis/snap-camera-server/wiki/Updating-Snap-Camera-(Unofficial)) | [Snap Camera Assets](https://github.com/ptrumpis/snap-camera-server/wiki/Snap-Camera-Assets)

Other lenses need to have their "Fallback Mode" patched.
You can apply the fix with this tool: [Snap Lens Tool](https://github.com/ptrumpis/snap-lens-tool).

---

### Snap lenses with missing assets:
- [Aging](https://lens.snapchat.com/d6ef78e959484752b1122a7682c7a815)
- [Beauty Cartoon](https://lens.snapchat.com/ed7ecb7fd95f4b309de41b194323b329)
- [Big Jaw](https://lens.snapchat.com/fb04e2f6ee114a87aa257f84a62c2e9b)
- [Caricature](https://lens.snapchat.com/a2425289059644698fcdf8797909f4c6)
- [Cartoon Kid](https://lens.snapchat.com/96d8b6d6194c4d71a0f8189c0fa6e59a)
- [Cartoon Punk](https://lens.snapchat.com/64a8e951d4ff4faba84085521030f745)
- [Chubby face](https://lens.snapchat.com/05914751ef514a969f0635c6861a7071)
- [Confident Face](https://lens.snapchat.com/430752224480442686ffd551bcf90340)
- [Cry Baby](https://lens.snapchat.com/48c94883620a484f8f7e3960a967d611)
- [Crying](https://lens.snapchat.com/924f374625604452a5d1273dad9d9726)
- [Cute Anime](https://lens.snapchat.com/aa2b5608536a40c69c818ecb4a1afecd)
- [Dashing](https://lens.snapchat.com/dcb77b4d79854802b3a8a8fca4ea315c)
- [Devilish Smile](https://lens.snapchat.com/11569557967e47f4b1d0592ee639740a)
- [Doll Face](https://lens.snapchat.com/6a07e8eae8f74001b2a675dc47ee3dee) | [Doll Face](https://lens.snapchat.com/6547b451225945b7bf681a1b1d1ebb26)
- [Elf Baby](https://lens.snapchat.com/c1988a4f4a994304898a52a80c703911)
- [Evil Beauty](https://lens.snapchat.com/b424e87885e74950a88b1f940a648c01)
- [Evil Grin](https://lens.snapchat.com/b3b0541431b948f185f9d6d30d1bbc28)
- [Fabulous Clown](https://lens.snapchat.com/202048517c29462ba3d69c2953a5d176)
- [Flabbergasted](https://lens.snapchat.com/6a18d81e3f2f433c9bb51545daf1519e)
- [Freckles and Brows](https://lens.snapchat.com/b3da4e171fd3478d9a93e5ac3941b460)
- [Frown Face](https://lens.snapchat.com/e078140698f041f7b202135bdee57f94)
- [Funny Face](https://lens.snapchat.com/439e9a5358ac48419bb5fe63f604145c)
- [Gloom](https://lens.snapchat.com/aa4fa84bfc444ba087578bbcb00fb93d)
- [Glowing Doll](https://lens.snapchat.com/5ff1b3ec1d2c4a34bb13a7ae5f308afb)
- [Good Look](https://lens.snapchat.com/13bb981e973f46439bdb1bd22383d568)
- [Goofy Gaze](https://lens.snapchat.com/bae79989338c485388e4c24f833268ae)
- [Grinning Face](https://lens.snapchat.com/e1937e7ab63c4462b728b0f13408aa8b)
- [Grumpy Kid](https://lens.snapchat.com/54b7dbfa8b284e6fadf3e78218affae2)
- [Intense Charm](https://lens.snapchat.com/c389ffee0fe64127950b64c9d987bdc4)
- [Intense Look](https://lens.snapchat.com/7c661e3bd983492ba54485d61473e3cb)
- [Loud Smile](https://lens.snapchat.com/4c37d1b37c72455790213b7604b513ac)
- [Marvelous](https://lens.snapchat.com/df618867335a42b1abfda6eab21d2ffb)
- [My Avatar](https://lens.snapchat.com/eab88571198f468c856fbc1513d80f0e)
- [Old Cartoon](https://lens.snapchat.com/f10ac1da02f5480f986d7753e9ab5faf)
- [Open Mouth](https://lens.snapchat.com/907352cc02ed4397a8aceb81b993cd61)
- [Pearly Whites](https://lens.snapchat.com/3a97d9f933144cb190ba448649f97b20) | [Pearly Whites](https://lens.snapchat.com/1e69589f33de42fe8fade9c4ea3b6773)
- [Pointy Nose](https://lens.snapchat.com/84dff5b8255d4cd8bd8db14621b7b37c)
- [Polished](https://lens.snapchat.com/f9990976ab5e4ae59cfea5d292084964)
- [Queen Disgust](https://lens.snapchat.com/c7a92bc23a5942f49803f21781001510)
- [Radiant](https://lens.snapchat.com/4207f3f949b84592845e1d04c23047b8)
- [Sad Punk](https://lens.snapchat.com/6c41a686e4ac4a3da07d9f1c1a8d69bf)
- [Scary Smile](https://lens.snapchat.com/ce55d7f2d6774981a729fa3330e68a93)
- [Serious Cartoon](https://lens.snapchat.com/d0587f8667bd46c28628d2d290a13b8d)
- [Shook](https://lens.snapchat.com/a4a202445b994be4924b59064621d661)
- [Sibling](https://lens.snapchat.com/fa9b5bebd896495c80f78393eb7209e4) | [Sibling](https://lens.snapchat.com/61d4eb8740a34deeb22648b041d8e306)
- [Silly Smirk](https://lens.snapchat.com/5a4ffe7f3420413a84abc326701cd736)
- [Smile Face](https://lens.snapchat.com/4ce480b62e704f04aed887a4a3f25819)
- [Soft Make Up](https://lens.snapchat.com/b1bc399b6c1b41f3bf97eae86960048e)
- [Stone Face](https://lens.snapchat.com/8638074fef7f42c78eb749da32f85270)
- [Strange Face](https://lens.snapchat.com/eec94b255dd54d0e8d730ca8448f6739)
- [Stumpy Look](https://lens.snapchat.com/7421db631e484b2787556498fa154ff2)
- [Stunning Beard](https://lens.snapchat.com/269071f8d4c041f6b8bfbd7572c8f500)
- [Stunning Look](https://lens.snapchat.com/a86b8e14bc6547bd81570de7a2351e31)
- [Sun-kissed Charm](https://lens.snapchat.com/8fdaf8054bf44ea8b863999ae98a767a)
- [Tongue Out Furious](https://lens.snapchat.com/5fa34a895db94234853dc61a6b30da12)
- [Tongue Twister](https://lens.snapchat.com/92b605bca6a04e78b93664797a770b38)
- [Unhinged](https://lens.snapchat.com/1eb7c9591f7e49c6a262dad46dd72b03)
- [Very Angry](https://lens.snapchat.com/00ceb5f848514274bf27cbd79f733f92)
- [Vintage Poster](https://lens.snapchat.com/40ca3a10a15345ed87bc05e30f31a661)

---

These lenses need to have their "Fallback Mode" patched with [Snap Lens Tool](https://github.com/ptrumpis/snap-lens-tool).

### Snap lenses that require further patching:
- [Aging](https://lens.snapchat.com/cea58064cd3c465b8fe95316eb23cded)
- [Angry Cartoon](https://lens.snapchat.com/83c892bd039a466182d8adee37022b92)
- [Astonished](https://lens.snapchat.com/baffa2b8dad749b48add4b5e4feec63c)
- [Baby Cheeks](https://lens.snapchat.com/2d086bb09e394a21a4d9639c3c6c738c)
- [Baby Eyes](https://lens.snapchat.com/208f37f03887468db799f4716bbf5904)
- [Big Head](https://lens.snapchat.com/a0daca382fd94cfebe03bc9cf4e7141e) | [Big Head](https://lens.snapchat.com/e7a106b6dc64409aaa106c46de44ae2f)
- [Big Lips](https://lens.snapchat.com/60b6af9cc9534bd7b70fda0fd0e4fb7c)
- [Bold Makeup](https://lens.snapchat.com/ebf297cd5a604a588cde58527903e23b)
- [Burgundy Makeup](https://lens.snapchat.com/7be2747864194ed187b110443b6045c9)
- [Caricature](https://lens.snapchat.com/3155d5aa568e4195a61331fbec12b94d)
- [Cartoon Kid 1](https://lens.snapchat.com/f4e3a48597f5442f92592cad20beb2ca)
- [Cartoon Kid 2](https://lens.snapchat.com/164fab7bbae74848a39b44436a83d64d)
- [Cartoon Kid 3](https://lens.snapchat.com/d97bba7f22e440a1b8065d18f4d93511)
- [Cartoon Kid 4](https://lens.snapchat.com/e033933e8d8b4f45b672498127761816)
- [Cartoon Kiddo](https://lens.snapchat.com/fd9ae9b9050c4264b793f4d84090f1bd)
- [Cartoon Lips](https://lens.snapchat.com/d729a7b6ca184fd498012963af43e207)
- [Cartoon Mustache](https://lens.snapchat.com/d67b030be4c54b3abe16bff18eb69bdc)
- [Chirp](https://lens.snapchat.com/8cbd878cb58a401cbf58c5f2a2387d2f)
- [Comical](https://lens.snapchat.com/2d017ddb60a2496b9242a5735c008d04)
- [Creature Cartoon](https://lens.snapchat.com/a2b24bd6163340be91b63116119ec234)
- [Dark Elf Character](https://lens.snapchat.com/0e1ae559a4fc4e2fa956a5da8e264411)
- [Disgust](https://lens.snapchat.com/41a72e9ae9a44f8891327ce6cf1c344d)
- [Dissatisfied](https://lens.snapchat.com/078485e4f4934ca7851657c41ab5b195)
- [Doll Style](https://lens.snapchat.com/2c17d554f45940c1a90bdbf8f5c4dd75)
- [Dwarf Character](https://lens.snapchat.com/f64332c03cdc4a29846e6a0c3d633fff)
- [Elf Character](https://lens.snapchat.com/8663335493004973b43192624a02da27)
- [Evil Cartoon](https://lens.snapchat.com/3c419d682ada4543b6306fbe9995be03)
- [Freckle Charm](https://lens.snapchat.com/fec6f425f19344108aed8b2d9a5e6c98)
- [Funny Smile](https://lens.snapchat.com/57cafb3421be4f39b89131a560450c14)
- [Goofball](https://lens.snapchat.com/a8bdccad55b54e41a5ffc0da0930093c)
- [Goofy Gaze](https://lens.snapchat.com/caf74cb1bbdc49ff94c08ea04251a884)
- [Grumpify](https://lens.snapchat.com/583f8026b518490eab3e4b9f68e67cd8)
- [Grumpy](https://lens.snapchat.com/28a50e5dcd3741ae92dd1a02508e4f53)
- [Grumpy Kid](https://lens.snapchat.com/f1cbc58ff1d34451be8ffd2278501ce1)
- [Handsome Cartoon](https://lens.snapchat.com/fa524b16fb51487599dfda9a78707768)
- [Happy Man](https://lens.snapchat.com/18e9948a4eac4c5086c8da92547d39d1)
- [Intense Look](https://lens.snapchat.com/9600cc09aa674908b0bdfb890c565aca)
- [Magical Elf](https://lens.snapchat.com/ceba20a5b7e24300acbf9495dfccb1c8)
- [Mixed Feelings](https://lens.snapchat.com/8e38100ea47c40a7a00629182209ee56)
- [Mood of the Day](https://lens.snapchat.com/fd735e3dacb049f5bd6c98ac6cb9eeb6)
- [Mouth Portraits](https://lens.snapchat.com/3837139480aa467699c3adba34f6387b)
- [My Avatar](https://lens.snapchat.com/b88dd9fa47b74c789eac67e0fe86fe6b)
- [My Cartoon Character](https://lens.snapchat.com/e67f1c43ebe54e239e6ed99d75207559)
- [My New Twin](https://lens.snapchat.com/20e98eba60bd4718924a1823c29ced90) | [My New Twin](https://lens.snapchat.com/2897e0ee4e064728a6d259dca1a78f6c)
- [Old](https://lens.snapchat.com/3e4b9510d0594e8c83c7cedb142c788c)
- [Pink vs Noir](https://lens.snapchat.com/f715bb46514e4765a0fd9693dd6e32dc)
- [Puffy](https://lens.snapchat.com/1afafe005df544a28410678d16ba0b50)
- [Red Hair Cartoon](https://lens.snapchat.com/d9f69840c81f42c791b48a4589d4e55f)
- [Red Hair Cartoon 2](https://lens.snapchat.com/1c2f761704f042fe8fb3d566baf6b59d)
- [Sad Smile](https://lens.snapchat.com/2d2688e3879948e7ba1c53f8fd818e42)
- [Serious Cartoon](https://lens.snapchat.com/d233daecd7424af787c2c6c404afbfcd)
- [Shining Star](https://lens.snapchat.com/754e805239bc43bfa93bc6e65643e627)
- [Sibling](https://lens.snapchat.com/33ef9032925d413c82f5cc619675adce) | [Sibling](https://lens.snapchat.com/6929b45285fc44b5b4492ba33c8aa56b)
- [Side Eye 🙄](https://lens.snapchat.com/2d90eec8df444daa8bb145269abf59f9)
- [Side Eye Grin](https://lens.snapchat.com/4adee3cdf847467f8af71b4df7e20d57)
- [Silly Smirk](https://lens.snapchat.com/d8c548f90a72422c87389393146637a6)
- [Stunning Beard](https://lens.snapchat.com/eccceeaf3f404bd683e6af6eb7e6e49c)
- [Stunning Look](https://lens.snapchat.com/feb35d2bf80b44ed8941e9099101285f)
- [Striking Cheekbones](https://lens.snapchat.com/48cb47dcef9e424882a76f8c06ae7a14)
- [Sunset Makeup](https://lens.snapchat.com/491b02605519412cad4f8d3b033bb66d)
- [Surprised](https://lens.snapchat.com/4f02649cb28e4b7393d86ea1b4450767)
- [Surfer Charm](https://lens.snapchat.com/7acae126e57245b4900573bfbeb8f6ba)
- [Sweaty face](https://lens.snapchat.com/c737a54ef37e4225bb485e8c7f2dda4d)
- [Tired-Eyed](https://lens.snapchat.com/ceaa8c3d774f438184d71779a15a0e06)
- [Tongue Out Furious](https://lens.snapchat.com/11fa0c8f24024b6baae85d232ccb72c7)
- [Uncanny Smile](https://lens.snapchat.com/16d500f8ab8b444a909d1c746f70a054)
- [Vibrant](https://lens.snapchat.com/88a23d136e8649ef9b645862d856fa07)

---

This list will be expanded in the future with your help.

**Comment by DermXZ** [comment]
https://lens.snapchat.com/e3c2f7824b6e482285c2b58e30a08ec8?sender_web_id=9b5c4337-f382-4361-b0c9-efde83a34cd4&device_type=desktop&is_copy_url=true

**Reply by ptrumpis** [reply]
solved with an unoffical update (read wiki)

**Comment by sverit** [comment]
https://lens.snapchat.com/2d086bb09e394a21a4d9639c3c6c738c
https://lens.snapchat.com/2897e0ee4e064728a6d259dca1a78f6c
https://lens.snapchat.com/20e98eba60bd4718924a1823c29ced90
https://lens.snapchat.com/6929b45285fc44b5b4492ba33c8aa56b
https://lens.snapchat.com/269071f8d4c041f6b8bfbd7572c8f500
https://lens.snapchat.com/b88dd9fa47b74c789eac67e0fe86fe6b
https://lens.snapchat.com/11fa0c8f24024b6baae85d232ccb72c7
https://lens.snapchat.com/feb35d2bf80b44ed8941e9099101285f
https://lens.snapchat.com/41a72e9ae9a44f8891327ce6cf1c344d
only shows a green dotted line for aligning my face for me

**Reply by ptrumpis** [reply]
Thanks, I updated the list.

**Comment by ghalta22** [comment]
I've messed with a few of the listed non-working lenses and was able to get them running successfully.

Unfortunately, it required modification to the lens (and thus unpacking + repacking). There is a javascript file inside (e.g. File366.js) with a `setLensMode(isFallback)` function. A bunch of possible ways to patch, but simplest is changing `var defaultValue = true;` to `var defaultValue = false;` at the top of that file. (Also worth noting, in the ones I've looked at, this is already set automatically for macos. So possible these lenses do work as-is on macs).

Not sure if there's a cleaner way to do this, there appears to be some sort of configuration system that this default value is used with if not set. If we could hook into that, or access and modify the javascript environment (or even override the detected platform or device model), a more generic fix might be possible without needing to patch individual lenses.

**Reply by ptrumpis** [reply]
Intressting!
- I could try to re-pack the lenses on the server side and serve the re-packed file to the client application.
- Or maybe a batch patching tool for the lens cache folder.

Very well done!

**Reply by k0d3rz** [reply]
Hi @ptrumpis & @ghalta22 , I am curious on how to get the Snap Camera client application to validate & use an updated lens after unpacking & repacking it.
I tried @ghalta22's approach, and used "My New Twin" lens (id **60848631001**) to test it out.
- I extracted the zstd off the file
- decompressed it
- the js file that happened to have the **setLensMode** settings was also **File366.js**, where I updated **defaultValue** initialization to **false**
- I re-compressed the files to zstd with level 2 compression (seemed to be the closest to the original compression level used)
- I merged the zstd with the original header, and made sure to update that header with the updated zstd size
- I replaced the original **lens.lns** in "**\Snap Camera\cache\lenses\60848631001**" with my updated one and made sure **settings.json** had the lens in its **cachedInfo**
Then tried to use that lens on the client application but it didn't work and the log returned:
`Can't create lns archive:  "[LZDecompressor] decompressAll; Parse archive error; checksum: 7eb9e469022e8eb6e9c2a198f158ac490bf93437c70f22468b367f76856a0aa5"`
Although I have the signature patched version, and it works perfectly fine on any lens from the web (except the incompatible ones of course), it doesn't seem to work on my updated lens.
Am I supposed to generate a valid signature for my lens although the client application is signature patched? Am I missing an important step in order for it to work?
Attached is **lens.lns** after being updated (it is zipped for the sake of this upload since .lns extension is not accepted):
[lens.zip](https://github.com/ptrumpis/snap-camera-server/files/12609506/lens.zip)

Note: I even tried to host the updated lens on my internal server with a new unique id, same result.

**Reply by k0d3rz** [reply]
Never mind, I got it to work. I had updated the **Zstd size** field of the file's header, but didn't update the **Compressed size** field (@ 0x1C) although both fields carry the same value. After updating **Compressed size** too it works perfectly fine.

**Reply by ptrumpis** [reply]
### Comment on your steps
I never tried the process myself, but judging by your steps you would have to rebuild the header "a little bit" more:
- The size of the zstd blob changed (which you updated according to your steps)
- But the size of the unpacked file File366.js also changed, the original header would still hold the original file information of File366.js and this would affect multiple offsets inside the header

Hex offsets that would have changed:
- 0x0018 Size of all files inside this archive uncompressed
- 0x001C Size of all files inside this archive compressed
- And also the entire file allocation table starting at 0x0048 would need a rebuild.

---

### How I think it should be done (untested)
The best way in my opinion would be to upload the zip package with modified files to the server, not the LNS file
The client would download the zip and automatically convert it to a LNS file inside your cache

If you prefer to have a re-packed LNS file there is a python tool https://github.com/cvirostek/snapchat-lens-blender-io/

---

Edit:
Seems you were faster than me @k0d3rz.
Your file allocation table might still be off by 1 single byte though.

**Reply by k0d3rz** [reply]
@ptrumpis thanks for the prompt response! Yes, I realized minutes after my original post that there are actually 2 entries in the header that define the size of the zstd and I had updated only one. Regarding the rest, I made sure to keep the file structure and uncompressed file size exactly the same as the original after my update.
I will try your suggestion of hosting a zip package rather than lns.

**Reply by ptrumpis** [reply]
I created [Snap Lens Tool](https://github.com/ptrumpis/snap-lens-tool) to automatically fix the fallback mode.

**Comment by IZeeLikeEyeZee** [comment]
https://lens.snapchat.com/3837139480aa467699c3adba34f6387b 
https://lens.snapchat.com/754e805239bc43bfa93bc6e65643e627 (also seems like any new phil watson lenses aren't working)
not working for me, I did the unofficial update too

**Comment by Orpheusll** [comment]
I've been keeping track of all the lenses that don't work by SnapChat, here is an updated, revised List: would be nice if one person can verify if these indeed are not working. Be careful of similarly named ones not by snapchat.

Intense Look
Goofy Gaze
Disgust
Sibling
Mood of the Day
Funny Smile
Cartoon Mustache
Dark Elf Character
Dwarf Character
Elf Character
Strange Face
Surprised
Uncanny Smile
Comical
Silly Smirk
My Cartoon Character
Baby Cheeks
Dissatisfied
Aging
Funny Morph
Tired-Eyed
Stunning Beard 
Vibrant 
Cartoon Lips
Sibling 
My New Twin 
Flabbergasted 
Pearly whites
Cartoon Kid 4
Cartoon Kid 3
Cartoon Kid 2
Cartoon Kid 1
Grumpy Kid
Puffy
Bold Makeup
Beauty Cartoon
Sad Punk
Stunning Look
Radiant
Polished
Burgundy Makeup
Evil Cartoon
Mixed Feelings
Dashing
Stunning Look
Intense Charm
Magical Elf
Big Lips
Big Jaw
Cartoon Style
Gloom
Evil Grin
Serious Cartoon
Grinning Face
Cartoon Punk
Caricature
Cartoon Kid
Baby Eyes
Doll Face
Devilish Smile
Grumpy
Elf Baby
Scary Smile
Big Head
Frown Face
Angry
Pink vs Noir
Sad Smile
Goofball
Creature Cartoon
Tongue out Furious
Pointy Nose
Surfer Charm
Cartoon Kiddo
Big-Eyed Kid
Vintage Poster
Smile Face
Freckles and Brows
Soft Make Up
Red Hair Cartoon
Cute Anime
Cry Baby
Handsome Cartoon
Confident Face
Sunset Makeup
Chirp
Caricature
Crazy Hairstyle
Good Look
Stone Face
Grumpify
Side Eye
Glowing Doll
Sun-Kissed Charm
Chubby Face
Doll Style
Beauty Cartoon
Striking Cheekbones
Sweaty Face
Old Uncle
Funny Face
Unhinged
Loud Smile
Serious Cartoon
Evil Beauty
Marvelous
Bored
Stumpy Look
Open Mouth
Fatigues Eyes
Doll Face
Nose 
Side Eye
My Avatar
Queen Disgust
Old Cartoon
Red Hair Cartoon
Happy Man
Fabulous Clown
Angry Cartoon

Note: @ghalta22 

If your workaround gets these lenses to work, please help us out. THANKYOU if you do!

**Reply by ptrumpis** [reply]
Thanks, I will check them and update the main post, some of them have an identical working alternative.

**Reply by Orpheusll** [reply]
Holy shit, you got alot of these to work. Thanks a tonne! You've given the entire snap camera community so much aid, much appreciated!

For your further convenience I noticed and wrote down some things:

1.Some of the ones you claim aren't working actually work for me:

Sibling lenses for example have a second copy available through search that work. If you'd like I can double check all of them for you and report it in a few days.


2. For the ones not working, I noticed that when you take a photo with them using the small icon on the left they actually work. For some reason (Or atleast thats my hunch) the AR isn't tracking on video format. Maybe there's an error being thrown while handling the video component. I don't know how much that would help you but it should atleast give you the hint that they are working, just not in video format. here is an example:
![image](https://github.com/ptrumpis/snap-camera-server/assets/140996616/67ed2055-3d6c-4aba-9dc8-d8c74d62df8c)

**Reply by ptrumpis** [reply]
> 1.Some of the ones you claim aren't working actually work for me:

Thanks for the feedback, I will re-check them with a fresh snap camera installation.

**Reply by ptrumpis** [reply]
I cleared the list after re-checking them.

**Comment by ptrumpis** [comment]
There is now a new tool available that should automatically fix the non working lenses.
It's still in early development and you need to manually upload the patched lenses to the server.

[Snap Lens Tool](https://github.com/ptrumpis/snap-lens-tool)

**Comment by linuxgeek12** [comment]
i am trying to get Pumpkin Head lens to work... https://snapchat.com/unlock/?type=SNAPCODE&uuid=e9fcc3538e524da189ae6deaefef2427&metadata=01 
i was able to compile and run the Snap Lens Tool but not sure where to go from there.  tried to run Set Fallback Mode on the lens in question, Expected to find the .lns file in /Snap Camera/cache/lenses but i do not .

**Reply by ptrumpis** [reply]
I will update the lens tool soon to be more user friendly.
The server will also receive a big update to automatically patch incompatible lenses.

Right now everything is hackerman style:
1. You need to login to Adminer
2. Search for "Pumpkin Head" or the UUID e9fcc3538e524da189ae6deaefef2427
3. Search results should give you the lens ID
4. So the lens file is located at "/Snap Camera/cache/lenses/{LENS_ID}/lens.lns" in your cache
5. Copy the file to your server /import folder
6. Click "Fallback Mode" button and select the file inside the /import folder with snap lens tool
7. Use Adminer to update the lens_url (unlock table) and make it point to your patched file in the /import folder
8. Everything inside your /import folder will be available under http://studio-app.snapchat.com/import/

[This Video](https://www.youtube.com/watch?v=alo49et3QxY&t=572s) is a little outdated but might help.

**Reply by ptrumpis** [reply]
Lens ID: 58130770949

But it works on my system without patching.

**Reply by linuxgeek12** [reply]
Thanks! that's super helpful and the video does still help.

Since it's working on your system it's obviously not the lens and must be something specific with my 

when i try to select the lens it never loads in snap camera, just a circle that spins and spins. (so far only happening to this one lens). 

I noticed that for this lens it never makes a {LENS_ID} folder in the /cache/lenses folder and if i take a look at the log located in /Snap Camera folder it says...  
"Checksum doesn't match for lens archive: 58130770949"

**Reply by ptrumpis** [reply]
> "Checksum doesn't match for lens archive: 58130770949"

You need to apply a [signature patch](https://github.com/ptrumpis/snap-camera-signature-patch).
The message indicates that no signature patch is currently applied.

It's similar to an iPhone Jailbreak.

**Reply by linuxgeek12** [reply]
Yes, that was it. applied the patch, super easy.  Everything works now!

**Comment by heelee912** [comment]
https://www.snapchat.com/lens/c98355cd57274490888c1d6765bc5460?type=SNAPCODE&metadata=01
https://www.snapchat.com/lens/a0d281e376bb41c482009de25909d74c?type=SNAPCODE&metadata=01

I Updated Snap Camera (Unofficial) and those two are keep loading.

**Comment by sissh** [comment]
Hello, first thank you for keeping snap camera alive.
This filter doesn't seem to work on my end, despite getting the unofficial update and the assets as well as patching fallback mode. It only shows a transparent rectangle over my face.

Tough Guy Face :

https://www.snapchat.com/lens/353a6b89caa74252aaa930d172da7ea4?type=SNAPCODE&metadata=01

[Log](https://github.com/user-attachments/files/16041629/Log.txt)

**Reply by LtCzr** [reply]
pls somebody fix this, i really want this filter

**Reply by thynomythe** [reply]
Hey did you get this filter? I want to preserve it for a future project.

**Reply by sissh** [reply]
> Hey did you get this filter? I want to preserve it for a future project.

last I checked it didn't work, but I tried it more than a year ago.

**Comment by LunaHackz** [comment]
Hey all, thanks for all the time and effort you guys have put into this project. For the life of me, I cannot get this lens to work.

https://www.snapchat.com/lens/0bfb145fa7b4445198821eaefc0fa7f6?type=SNAPCODE&metadata=01

I have tried all the steps and workarounds I have found but it doesn't load. Could someone look at it for me and see if maybe I just did something wrong? 

There are a few comments at the bottom of the log where I attempted to use the lens but I don't know what they mean. 

[Log.txt](https://github.com/user-attachments/files/17020037/Log.txt)

Thanks so much!!

**Comment by Jooeellee** [comment]
Yes long time ago

Get Outlook for Android<https://aka.ms/ghei36>
________________________________

**Comment by gabonacorp** [comment]
Hi everybody,

I am hardy try to make this lens work: https://www.snapchat.com/lens/a375b6d69d0c4feda05a84e0ab58e471?sid=a469e440385c4d19b03a5f690983015c&invite_id=EI72yt2P

I have the signature patched snap camera. at first time the log sais that the it failed to download an asset named RND-802_fashion_senior_v1. Luckily it can be downloaded from the snap camera assets so I did it and copied it to the PrecachedAssets.bundle folder. After this snap camera recognized the missing asset and from that no error comes up. In snap camera app the lens looks like to load correctly, the two camera feeds shows up, the loading signal stops but the aged results is not coming up .
I would really appreciate any help in this matter.

thanks,

gabona

**Comment by afiazk** [comment]
Hi everyone, ill give 10$ cash for person who can fix this

i really need to use this lens below, for the person who can fix the problem i will give u some cash.
the problem is : webapp-1    | [Info] 😕 This lens cannot currently be activated: 56770180889

Lens : https://www.snapchat.com/lens/88d26d72bca34469ac80407b59562ce3

@ptrumpis or the people can u help me? i really need this lens, this lens is keep loading on the any version windows!

**Reply by afiazk** [reply]
i already buy the assets but i dont know it loaded or no, i dont know how to use correctly, can u pls help me to fix this ill give u cash 10$ @ptrumpis

**Reply by alexequilibrium9-creator** [reply]
try to make [unofficial update](https://github.com/ptrumpis/snap-camera-server/wiki/Updating-Snap-Camera-(Unofficial))
try use different versions of Lens Studio... May be it will help you

**Reply by DKogut17** [reply]
I'm confused is the patch supposed to allow users to use the search bar in Snap Camera to paste URL links to the lens they want to use? Because mine is definitely not doing that.

**Comment by tempeststudio-git** [comment]
Not sure if this is because it's incompatible or something else, but has anyone been able to use "locked" filters? In particular, this lens:
https://www.snapchat.com/lens/de4333d8e418462bb5319b9fefca76ae

On Snap Camera, it loads up the base lens fine (a potato), but then shows a bunch of other alternatives all locked with "Unlock for xxx" - seems to be some sort of in-app currency??? Those locked lenses do load fine, but are blurred so you can't see your face.

Is there a workaround to unblur or unlock those additional lenses? 

Appreciate any help! @ptrumpis

**Comment by Seiko19950818** [comment]
May I ask how to use lens? Most of the lens are not available. When I select those lenses, they are shown:

webproxy-1  | 172.18.0.1 - - [28/Apr/2026:10:01:47 +0000] "GET /vc/v1/explorer/unlock?uid=35734160986 HTTP/1.1" 499 0 "-" "SnapCamera/1.21.0.0 (Windows 10 Version 2009)" "-"webapp-1    | [Timeout] (1/2) https://archive.org/wayback/available?timestamp=20230601000000&url=lens.snapchat.com%2Fe971b3e506424e748b0aa44add08353d*webapp-1    | [Timeout] (2/2) https://archive.org/wayback/available?timestamp=20230601000000&url=lens.snapchat.com%2F1194bb7952e94c769041c9be86b32993
